### ISSUE TITLE
(Discussion) Tasks: Reduce the Task related quick open handlers

### ISSUE THREAD

**Issue by egamma** [Microsoft/vscode#29616]
Tasks currently contributes different quick open handlers for:
- build task
- test task
- task
- terminate task
- restart task

Now that we have a Task menu do we still need all the quick open handlers? 

In particular it confused me when I ran "Task>Run Build Task" and I got presented with the 'build task' quick open handler:
![image](https://user-images.githubusercontent.com/172399/27591780-1079872a-5b53-11e7-808a-8a1b64f77b18.png)

It appears that I'm presented with all detected tasks and not only the build tasks.
For the first time user it would be more intuitive to now show Quick Open but to show a Quick Pick with an explaining message as is done when the user picks a problem matcher.
![image](https://user-images.githubusercontent.com/172399/27591907-6dea28ec-5b53-11e7-995d-1af5d7fac810.png)

**Comment by dbaeumer** [comment]
I see the point. The reason why I kept the quick open is that I am so used to quick open that I don't want to go back to a global menu. But it is a fair request for newcomers.

Regarding the build task: The listed tasks are all detected as build tasks since they contain build or watch in the task name. I relaxed that since we can mark a build task as primary and therefore bind it to Ctrl+Shift+B and assign a problem matcher. I found it for example confusing that compile or watch was not proposed in the past.
